### PR TITLE
Added tables to replace the introductory part on core properties

### PR DESCRIPTION
**Pull request by iherman**
This is an alternative to #519 (but taking into account the discussions on that PR) and an attempt to get #404 resolved.

This PR replaces most of the introductory section on Core Properties by three tables that gives an overview of all the properties defined in the spec, with associated usage and value constraints. As an editorial result, the sectioning of the full §5 has changed somewhat.

I did not make any change on the more detailed specification of the properties (i.e., the subsection of §5.2 in the new structure); this would indeed be better to be done by one of the regular editors of the document. As a result, there is now overlap/redundancy in the specification with the same constraints defined at two places. I am actually not sure this is a bad thing, in case it makes the specification more readable.

(As a minor point, I have anticipated the merge of #522 and the `alsoKnownAs` value is an ordered set in the table. I have not touched the detailed definition, merging #522 should take care of that.)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/pull/528.html" title="Last updated on Jan 6, 2021, 8:02 AM UTC (66173c5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/528/10744a0...66173c5.html" title="Last updated on Jan 6, 2021, 8:02 AM UTC (66173c5)">Diff</a>

### CODE REVIEW

**Comment by iherman** [comment]
A question on the constraints. In example 22 on various service endpoints there is one that has, beyond the (required) `serviceEndpoint` property (and a number of non-standard properties) that uses the `verificationMethod`, which is a core property. The possibility of using this is missing from the table (and is missing from the text, hence I missed it!). 

Before I add it to the table, it is not clear whether the value, when in a service endpoint, can be a full-blown verification method object or only a URI. Which is it?

**Review by rhiaro — COMMENTED** [review]
This looks great, thanks Ivan! 

With the new nesting, I think there are going to be some brutal merge conflicts with other open PRs. But without it, I don't think there will be any. I prefer not nesting the "Detailed.." section anyway, not only for this reason..

I'd like to also do a pass to make sure duplication is removed, but maybe that can wait until after this is merged to make it easier. 

> Before I add it to the table, it is not clear whether the value, when in a service endpoint, can be a full-blown verification method object or only a URI. Which is it?

I think it can be either.. There is that whole bit about including verification methods by value or by reference ([under the example](https://w3c.github.io/did-core/#example-14-example-verification-methods)).

**Comment by iherman** [comment]
> With the new nesting, I think there are going to be some brutal merge conflicts with other open PRs. But without it, I don't think there will be any. I prefer not nesting the "Detailed.." section anyway, not only for this reason..

Let us discuss this further... I am not bound to that nesting, of course, and I understand the PR merge issues but, I believe the structure of the document is clearer if we do that. I am happy to yield to the editors and you on this, though.

**Comment by iherman** [comment]
> I'd like to also do a pass to make sure duplication is removed, but maybe that can wait until after this is merged to make it easier.

I would love if you did that, I did not want to modify the text there. But, indeed, it may be a good idea to merge first and do it in a second pass to avoid some git problems.

File: index.html
Comment (iherman) [reply index.html]: Resolved in https://github.com/w3c/did-core/pull/528/commits/50db190fda6cecf92bb5700593d1639920a55428

File: index.html
Comment (iherman) [reply index.html]: Resolved in https://github.com/w3c/did-core/pull/528/commits/50db190fda6cecf92bb5700593d1639920a55428.

File: index.html
Comment (iherman) [reply index.html]: Resolved in https://github.com/w3c/did-core/pull/528/commits/50db190fda6cecf92bb5700593d1639920a55428.

File: index.html
Comment (iherman) [reply index.html:1514]: See https://github.com/w3c/did-core/pull/528#issuecomment-754554022.

**Review by talltree — APPROVED** [review]
I am good with the substance of this PR. I defer to Ivan and Amy to work out the formatting questions.

**Review by msporny — APPROVED** [review]

**Comment by iherman** [comment]
@msporny instead of going through a conflict resolution here (which would have turned out to be horrific and error-prone) I re-created #536 transferring the changes on the latest version of the main branch.

Cc @rhiaro

**Comment by iherman** [comment]
Instead of handling the complex PRs, a new one has been created (#536). @msporny this PR should be closed without merge.

**Comment by msporny** [comment]
PR #536 supersedes this PR, closing per request from @iherman.

**Pull request closed without merging**
